internal/services/firewall/firewall_resource.go

```diff
@@ -578,6 +616,33 @@ func resourceFirewallDelete(d *pluginsdk.ResourceData, meta interface{}) error {
 	return err
 }
 
+func expandAutoscaleConfiguration(d map[string]interface{}) (azurefirewalls.AzureFirewallAutoscaleConfiguration, error) {
+	configuration := azurefirewalls.AzureFirewallAutoscaleConfiguration{}
+	minPresent := false
+	maxPresent := false
+	min := 0
+	max := 0
+	if _, ok := d["min_capacity"]; ok {
+		minPresent = true
+		min = d["min_capacity"].(int)
+	}
+	if _, ok := d["max_capacity"]; ok {
+		maxPresent = true
+		max = d["max_capacity"].(int)
+	}
+	if minPresent && min > 0 {
+		configuration.MinCapacity = pointer.FromInt64(int64(min))
+	}
+	if maxPresent && max > 0 {
+		configuration.MaxCapacity = pointer.FromInt64(int64(max))
+	}
+	_, errors := validate.AutoscaleConfiguration(d)
```

Comment:
Instead of validating the values here, could you please add it to the `CustomizeDiff` field of the schema definition, so that users can get the validation error during plan time?

Reference: https://github.com/hashicorp/terraform-provider-azurerm/blob/7646a3ce465242c7aef716aefd475677657be2a1/internal/services/cdn/cdn_frontdoor_profile_resource.go#L99